llama_index/objects/base_node_mapping.py

```diff
@@ -90,3 +141,40 @@ def to_node(self, obj: Any) -> TextNode:
 
     def _from_node(self, node: BaseNode) -> Any:
         return self._objs[hash(node.get_content(metadata_mode=MetadataMode.NONE))]
+
+    def persist(
+        self,
+        persist_dir: str = DEFAULT_PERSIST_DIR,
+        obj_node_mapping_fname: str = DEFAULT_PERSIST_FNAME,
+    ) -> None:
+        """Persist object node mapping.
+
+        NOTE: This may fail depending on whether the object types are
+        pickle-able.
+        """
+        if not os.path.exists(persist_dir):
+            os.makedirs(persist_dir)
+        obj_node_mapping_path = concat_dirs(persist_dir, obj_node_mapping_fname)
+        try:
+            with open(obj_node_mapping_path, "wb") as f:
+                pickle.dump(self.__dict__, f)
```

Comment:
Any reason to not specifically do `pickle.dump(self)` here? It would make the loading section a little more simple.

Reply:
yeah, I was trying to do this to hold on the possibility that other subclasses may be able to implement `persist`. But in accordance to your other comment, if we do remove the others (as there really is no need to check them as we know they're not implemented), then we can make this simpler too. I'll do that -- thanks!

Reply:
oh hang on, I think I had a mix-up. My bad, you can disregard my earlier comment. Let me try this simpler method out.